source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+| Official Sensu Support for the Tool                 | High   | High                | Low               | Low      | N/A                   |
+| Reproducibility                                     | High   | High                | High              | High     | N/A                   |
+| Easy of use getting started                         | Medium | Medium              | High              | Medium   | Low                   |
+| Language extensibility                              | High   | High                | Low               | Low      | N/A                   |
+| Separation between config data and code             | Hiera  | Databags/Attributes | Random variables? | Pillar   | N/A                   |
+| Module re-usability?                                 | High   | High                | Low               | Low      | None                  |
+| **Review of the Sensu Module/Cookbook/Etc**         |        |                     |                   |          |                       |
+| Version of the module Used                          | 1.5.5  | 2.10.0              | 0.1.0             | c6324b3  |                       |
+| Sensu Module Feature Completeness                   | High   | High                | Medium            | Medium   |                       |
+| Sensu Module Integration with Other Modules         | None   | Extreme?            | None              | None     |                       |
+| Sensu Module Flexibility                            | High   | High                | Medium            | Low      |                       |
+| Sensu Module Re-usability                           | High   | High                | High              | Low      |                       |
+| How Opinionated Was It?                             | Low    | High                | Low               | Medium   |                       |
+| Usability with Sensu's Embedded Ruby                | Yes    | Yes                 | Not natively      | Sorta    |                       |
+
+## Conclusion
+
+The way I see it, there are two camps. Chef and Puppet both provide a rich
+language to build modules with. For example, the [PuppetLabs RabbitMQ module](https://forge.puppetlabs.com/puppetlabs/rabbitmq) contains all the code
+to interact with RabbitMQ. The main Puppet codebase doesn't know anything about
+RabbitMQ. The same goes for Chef. Both Chef and Puppet also have their own DSL.
+Puppet uses yaml files for Hiera, but they are for config only, unlike
+Ansible/Salt.
+
+In the other camp is Ansible and Salt. They have a simplified config language,
+and require the help from the core software to do the "heavy lifting" of the
+raw types. For example, the Salt [RabbitMQ formula](https://github.com/saltstack-formulas/rabbitmq-formula/blob/master/rabbitmq/config.sls#L29)
+requires the help of [core Salt RabbitMQ module](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.rabbitmq.html)
+to provide the primitives.
+
+### Final Thoughts
+
+* Puppet
```

Comment:
I think it's super important that every "final thought" is introduced in some way in the post.  I'm adding some notes to that end.